.github/workflows/build.yml

```
@@ -1,45 +1,157 @@
 name: Build
 on:
   push:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 jobs:
-  build:
+  prt-contracts:
     runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-24.04
```

`latest` is a moving target (a slow moving one) and things can break in the future.

It's good practice to use a fixed version.

This happens at other places in this PR.